api/src/test/java/org/openmrs/module/locationbasedaccess/aop/UserSearchAdviserTest.java

```diff
@@ -0,0 +1,617 @@
+        
+        List<User> userList = userService.getAllUsers();
+        assertEquals(userList.size(), 8);        
+        assertNotNull(userList.get(0));
+        assertEquals(userList.get(0).getUserProperty(DEMO_USER_PROPERTY),"");        
+        assertNotNull(userList.get(1));
+        assertEquals(userList.get(1).getUserProperty(DEMO_USER_PROPERTY),"");      
+        assertNotNull(userList.get(2));
+        assertEquals(userList.get(2).getUserProperty(DEMO_USER_PROPERTY),"");      
+        assertNotNull(userList.get(3));
+        assertEquals(userList.get(3).getUserProperty(DEMO_USER_PROPERTY),"");        
+        assertNotNull(userList.get(4));
+        assertEquals(userList.get(4).getUserProperty(DEMO_USER_PROPERTY),"");        
+        assertNotNull(userList.get(5));
+        assertEquals(userList.get(5).getUserProperty(DEMO_USER_PROPERTY),location1.getUuid());
+        assertNotNull(userList.get(6));
+        assertEquals(userList.get(6).getUserProperty(DEMO_USER_PROPERTY),location2.getUuid());
+        assertNotNull(userList.get(7));
+        assertEquals(userList.get(7).getUserProperty(DEMO_USER_PROPERTY),location1.getUuid());
+    }
+    
+    @Test
+    public void getAllUsers_nonAdminUserWithOutLocationPropertyShouldNotGetUserInformation(){
+        User normalUser = userService.getUser(DEMO_USER1_ID);
+        Context.authenticate(normalUser.getUsername(), "userServiceTest");
+        assertFalse(normalUser.isSuperUser());
+
+        assertEquals(normalUser.getUserProperties().size(), 0);
+        
+        List<User> userList = userService.getAllUsers();
+        assertEquals(1, userList.size());
+        assertNotNull(userList.get(0)); //Users can access themselves
+        assertEquals(userList.get(0).getUserProperty(DEMO_USER_PROPERTY),"");
+        assertEquals(userList.get(0),normalUser); //Users can access themselves
+        
+    }
+    
+    @Test
+    public void getAllUsers_nonAdminUserWithLocationPropertyShouldGetOnlyAccessibleUserInformation(){
+        User normalUser = userService.getUser(DEMO_USER2_ID);
+        Context.authenticate(normalUser.getUsername(), "userServiceTest");
+        assertFalse(normalUser.isSuperUser());
+
+        Location location1 = locationService.getLocation(DEMO_LOCATION1_ID);
+
+        assertEquals(normalUser.getUserProperties().size(), 1);
+        String locationUserProperty = normalUser.getUserProperty("locationUuid");
+        assertNotNull(locationUserProperty);
+        assertEquals(locationUserProperty, location1.getUuid());
+        
+        List<User> userList = userService.getAllUsers();
+        assertEquals(userList.size(), 2);
+        assertNotNull(userList.get(0));
+        assertEquals(userList.get(0).getUserProperty(DEMO_USER_PROPERTY),location1.getUuid());
+        assertEquals(userList.get(0),normalUser);
+        assertNotNull(userList.get(1));
+        assertEquals(userList.get(1).getUserProperty(DEMO_USER_PROPERTY),location1.getUuid());

```

Comment:
After this line, Have you proof that the user object is not the normalUser?
Actually it should be another user object to address this test case

Reply:
I have added 
assertNotEquals( userList.get(1), normalUser);
after line 201. I think that would verify that.